pytensor/scan/basic.py

```diff
@@ -598,6 +598,10 @@ def wrap_into_list(x):
 
                 # Try to transfer test_value to the new variable
                 if config.compute_test_value != "off":
+                    warnings.warn(
+                        "test_value machinery is deprecated and will stop working in the future.",
+                        FutureWarning,
+                    )
```

Comment:
These warnings show up in too many places (not talking about this file/line specifically)? Should only be when users set the config flag or manually assign a test_value / call a function that is specific about test values?

Reply:
The warning is raised whenever a user tries to do a ```x.tag.test_value``` and whenever in the internal code the ```config.compute_test_value != "off"```. Are you suggesting that instead of raising the warning when internal code has ```config.compute_test_value != "off"```, I raise the warning when user explicitly sets ```config.compute_test_value != "off"``` ?
In this way, the warning will appear at less places within the code but will be appropriately raised.

Reply:
Yes, we don't want it to become incredibly noisy. Just on the user entry points to tell them to stop using the machinery